control-plane/csi-driver/src/bin/node/nodeplugin_svc.rs

```diff
@@ -2,150 +2,43 @@
-
-    if let Some(device) = Device::lookup(&uuid).await.context(InternalFailure {
-        volume_id: volume_id.to_string(),
-    })? {
-        let device_path = device.devname();
-        if let Some(mnt) = mount::find_mount(Some(&device_path), None) {
-            let dest = mnt.dest.display().to_string();
-            let args = [freeze_op, &dest];
-            let output = Command::new(FSFREEZE)
-                .args(args)
-                .output()
-                .await
-                .context(Io {
-                    volume_id: volume_id.to_string(),
-                })?;
-            return if output.status.success() {
-                Ok(())
-            } else {
-                let errmsg = String::from_utf8(output.stderr).unwrap();
-                debug!(
-                    "{} for volume_id :{} : failed, {}",
-                    freeze_op, volume_id, errmsg
-                );
-                Err(ServiceError::FsfreezeFailed {
-                    volume_id: volume_id.to_string(),
-                    error: errmsg,
-                })
-            };
-        } else {
-            // mount::find_mount does not return any matches,
-            // for mounts which are bind mounts to block devices
-            // (raw block volume).
-            // It would be incorrect to return the VolumeNotFound error,
-            // if the volume is mounted as a raw block volume on this node.
-            // Use findmnt to work out if volume is mounted as a raw
-            // block, i.e. we get some matches, and return the
-            // BlockDeviceMount error.
-            let mountpaths = findmnt::get_mountpaths(&device_path).context(InternalFailure {
-                volume_id: volume_id.to_string(),
-            })?;
-            if !mountpaths.is_empty() {
-                debug!(
-                    "{} for volume_id :{} : failed for block device",
-                    freeze_op, volume_id
-                );
-                return Err(ServiceError::BlockDeviceMount {
-                    volume_id: volume_id.to_string(),
-                });
-            }
-            debug!(
-                "{} for volume_id :{} : failed, cannot find volume",
-                freeze_op, volume_id
-            );
-        }
-    }
-    Err(ServiceError::VolumeNotFound {
-        volume_id: volume_id.to_string(),
-    })
-}
-
-pub(crate) async fn freeze_volume(volume_id: &str) -> Result<(), ServiceError> {
-    fsfreeze(volume_id, "--freeze").await
-}
-
-pub(crate) async fn unfreeze_volume(volume_id: &str) -> Result<(), ServiceError> {
-    fsfreeze(volume_id, "--unfreeze").await
-}
-
 /// Lookup the device by its volume id.
-pub(crate) async fn lookup_device(volume_id: &str) -> Result<Box<dyn Detach>, ServiceError> {
-    let uuid = Uuid::parse_str(volume_id).context(InvalidVolumeId {
-        volume_id: volume_id.to_string(),
-    })?;
+pub(crate) async fn lookup_device(volume_id: &str) -> Result<Box<dyn Detach>, tonic::Status> {
```

Comment:
What's the reason for giving up the service error in favour of direct status?

Reply:
To reduce another layer of conversion from service error to status.

Reply:
I guess that's fine, though means you can't reuse this code for other internal/non-tonic things